endstate_correction/system.py

```diff
@@ -14,6 +14,26 @@
     check_implementation,
 )
 
+def gen_box(psf, crd):
```

Comment:
can you add a short docstring explaining what is happening in this function and add type annotations?